src/notebooks/R Mia Examples/Comparative-Metagenomics.qmd

```diff
@@ -0,0 +1,852 @@
+
+In the following examples we will show three popular ways of normalization: relative abundance, rarefaction and cummulative sum scaling.
+
+## Normalization by total sum scaling (TSS, relative abundance or proportions)
+The simplest way to normalize the differences in sample size is to transform the OTU counts table into relative abundance by dividing by the number of total reads of each sample. This type of normalization is also referred to as relative abundance or proportions. We use this normalization to compare taxonomic profiles, while alpha diversity indices are computed on the non-normalized matrix. The reason to do so is that we need a matrix of integer numbers as input.
+
+1. Transform taxonomy raw-counts matrix into relative abundance.
+
+```{r}
+filtered_tse <- transformAssay(filtered_tse, assay.type = "counts", name = "relabundance", method = "relabundance")
+
+```
+
+2. Agglomerate by class rank to the `glom_tse`.
+
+```{r}
+glom_tse <- agglomerateByRank(filtered_tse, rank = "Class", assay.type = "relabundance")
+
+```
+
+3. Get the top 15 features and store them in the `tss_tse`. We store the top features in a new tse object as keeping the `glom_tse` and `tss_tse` objects  will come in handy later on.
+
+```{r}
+# Take the top 15 features based on their means
+top15_tss <- getTopFeatures(glom_tse, method = "mean", assay.type = "relabundance", top = 15L)
+
+# Store those top 15 features and all relevant metadata in a new tse
+tss_tse <- TreeSummarizedExperiment(
+              assays = list(tss = assay(glom_tse, "relabundance")[top15_tss, ]),
+              colData = colData(glom_tse),
+              rowData = rowData(glom_tse)[top15_tss, ]
+            )
+```
+
+## Normalization by subsampling (rarefaction)
+Rarefaction is an alternative to relative abundance normalization to obtain an adjusted OTUs count matrix. The method is based on a process of subsampling to the smallest library size in the data set. The algorithm randomly removes reads until the samples reach the same library size. Despite the apparent disadvantage of discarding information from the larger samples, rarefaction is quite popular in microbial ecology. The first step is to find the smallest sample size. We can use the number of observed OTUs in the original matrix to do so.
+
+1. Find the smallest sample size.
+
+```{r}
+min_size <- min(colSums2(assay(filtered_tse, "counts")))
+```
+
+2. Rarefying to the smallest sample.
+
+```{r}
+filtered_tse <- subsampleCounts(filtered_tse, assay.type = "counts", size = min_size, name = "rarefy")
+```
+
+3. Agglomerate taxonomy at Class rank and save to `glom_tse`.
+
+```{r}
+glom_tse <- agglomerateByRank(filtered_tse, assay.type = "rarefy", rank = "Class", name = "rarefy")
+```
+
+4. Select the top 15 taxa and save them to `rarefy_tse`. We keep `glom_tse` and `rarefy_tse` separate as it will make things easier later. 
+
+```{r}
+# Get the taxa sums by row
+taxa_sums <- rowSums(assay(glom_tse, "rarefy"))
```

Comment:
scater::addPerFeatureQC if I remember correctly

Reply:
is that necessary do you think? is it not quicker just to save to a temporary variable than setting rowData to then reaccess

Reply:
I think both are OK here because there is no difference in terms of code length. However I would mention in the text that the scated::addPerFeatureQC function also calculates this together with some other stats so it can be handy